### PR TITLE
Feature/git

### LINKED ISSUE #3: Add the git interactions
This is postponed for now. If someone tries to add something like this, please look at what `npm version` does.

Something like the following:
A) Add git commit for the version bump in the Cargo.toml, optionally with a custom commit message.
B) Point a git tag named `v$major.$minor.$patch` at the commit made in A.

### PR DESCRIPTION
**Pull request by TomPridham**
i've added basic git functionality. all this does is make sure the working directory is clean, then commit the Cargo.toml changes and create a git tag at that commit. i'm not sure what else `npm version` does behind the scenes, but that seems like the bare minimum.

one thing i noticed was that on the next `cargo run/build` it will update the `Cargo.lock` file with the new version. i could take a stab at adding that functionality as well if you would like.

Fixes #3. not sure if there should be a follow-up issue to add more advanced functionality or if you want to update that issue

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -6,26 +6,33 @@ extern crate semver;
...
 
 use std::fs::{File, OpenOptions};
 use std::io::{Read, Write};
 use std::path::Path;
+use toml_edit::Document;
 
 use semver::Version;
 
 fn main() {
+    git::git_check();
```
Comment (wraithan) [inline src/main.rs]: git can't be required, there are other version control systems out there

**Comment by wraithan** [comment]
I like the PR with the exception of requiring git.

I don't want to break folks flows that happen to use this tool with the new feature doing things they don't expect. So starting by having it be an opt in option, and down the line with a major release flipping it to detect if there is a .git and have an opt out option after a suitable warning period would be ideal.

Thanks for sending this. I've been meaning to get back into this project, so maybe this will stir me into finishing the workspace support I started a while back.

**Comment by TomPridham** [comment]
would you like it to accept a flag to check whether to use git or just check for a `.git` folder at the same level as `Cargo.toml`? i think i've got the part checking for a `.git` folder working, but i'll have to confirm after work. what should the flag be, `--git`?

**Comment by wraithan** [comment]
`--git-tag` would probably be what I'd prefer. Checking at the same level as the `Cargo.toml` for the `.git` seems like it could have plenty of problems, especially in workspaces/monorepos. I'd rather omit the `.git` check for the initial opt in version. The test for cleanliness will capture the case where we're not in a git repo and someone tried to opt into the flag.

I mentioned checking for `.git` as a maybe in the next major version, it'd make sense to test for that and warn that in the next major version after that, tagging will become the default if in a git repo. I'm pretty sensitive to breaking folks flows with CLI tools, and even those this doesn't have a ton of users, I'd still rather play it safe.

Thanks for being amenable to my requested changes!

**Comment by TomPridham** [comment]
do you have a way to access flags in any of the files currently? i didn't see anything that looked like it did that, but wanted to make sure before i reimplemented something

**Comment by TomPridham** [comment]
i guess i should have looked at the clap docs first :grimacing:

**Comment by zkat** [comment]
very excited about this!!

**Comment by wraithan** [comment]
Me too, got some real life errands to attend to, then I'll be able to test this out :D

**Comment by TomPridham** [comment]
i was thinking of trying to add some tests, but once i started, i wasn't really sure what that would even look like since most of this is just calling out to git. do you think tests are necessary here?

**Comment by wraithan** [comment]
I'm fine with things as is, we could add tests that call out to git to make sure the appropriate things got created, which would, in a way, be testing our arguments to git. But I'm not super worried about it. My day unfortunately got away from me though and its gotten late.

I'll definitely get to this tomorrow during my lunch break at work.

**Comment by lukeschunk** [comment]
@TomPridham you're a baller 🏀

**Comment by zkat** [comment]
Yay

**Comment by wraithan** [comment]
I tested it locally, found one small issue `clippy` had with the `stdout.len() > 0` check and replaced it with a `!stdout.empty()` in a supplemental commit. Which points out I need to have `clippy` and `fmt` run in CI. This PR has been released as v1.1.0 of `cargo-bump` thanks again for taking the initiative to send this and listening to my feedback.

As I mentioned above, I used the `--git-tag` feature to make the 1.1.0 release commit and tag :D

**Comment by TomPridham** [comment]
awesome :) it's something i've been wanting myself, so it wasn't any trouble

**Pull request merged**